src/webgpu/shader/validation/expression/call/builtin/countLeadingZeros.spec.ts

```diff
@@ -0,0 +1,92 @@
+const builtin = 'countLeadingZeros';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import {
+  kConcreteIntegerScalarsAndVectors,
+  kFloatScalarsAndVectors,
+} from '../../../../../util/conversion.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  fullRangeForType,
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kValuesTypes = objectsToRecord(kConcreteIntegerScalarsAndVectors);
+
+g.test('values')
+  .desc(
+    `
+Validates that constant evaluation and override evaluation of ${builtin}() never errors
+`
+  )
+  .params(u =>
+    u
+      .combine('stage', kConstantAndOverrideStages)
+      .combine('type', keysOf(kValuesTypes))
+      .filter(u => stageSupportsType(u.stage, kValuesTypes[u.type]))
+      .beginSubcases()
+      .expand('value', u => fullRangeForType(kValuesTypes[u.type]))
+  )
+  .fn(t => {
+    const expectedResult = true; // countLeadingZeros() should never error
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      expectedResult,
+      [kValuesTypes[t.params.type].create(t.params.value)],
+      t.params.stage
+    );
+  });
+
+const kFloatTypes = objectsToRecord(kFloatScalarsAndVectors);
+
+g.test('float_argument')
+  .desc(
+    `
+Validates that float arguments are rejected by ${builtin}()
+`
+  )
+  .params(u => u.combine('type', keysOf(kFloatTypes)))
+  .fn(t => {
+    const type = kFloatTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ false,
+      [type.create(0)],
+      'constant'
+    );
+  });
+
+const kGoodArgs = '(1u)';
```

Comment:
@ben-clayton gave me feedback about how to improve these.
See discussion here on @toji 's PR
https://github.com/gpuweb/cts/pull/3497#pullrequestreview-1937798113 

For latest advice on how to organize and name cases, see

https://github.com/gpuweb/cts/blob/278383fad33d67229a7a592ce39762ae9f0addf5/src/webgpu/shader/validation/expression/call/builtin/radians.spec.ts#L77

Reply:
Done.